src/sentry/analytics/events/metric_alert_with_ui_component_created.py

```diff
@@ -1,14 +1,11 @@
 from sentry import analytics
 
 
+@analytics.eventclass("metric_alert_with_ui_component.created")
 class MetricAlertWithUiComponentCreatedEvent(analytics.Event):
-    type = "metric_alert_with_ui_component.created"
-
-    attributes = (
-        analytics.Attribute("user_id", required=False),
-        analytics.Attribute("alert_rule_id"),
-        analytics.Attribute("organization_id"),
-    )
+    user_id: str | None = None
```

Comment:
You mentioned that `user_id` should be an int, but sometimes the system treats it as a `str`. Should we stick with using `int` and convert it where needed, or is this something @constantinius will handle in a follow-up, as you mentioned?